tests/test_trainer.py

```diff
@@ -195,6 +197,26 @@ def forward(self, input_x, labels=None, **kwargs):
             loss = torch.nn.functional.mse_loss(y, labels)
             return (loss, y, y) if self.double_output else (loss, y)
 
+    class RegressionRandomPreTrainedModel(PreTrainedModel):
+        config_class = RegressionModelConfig
+        base_model_prefix = "regression"
+
+        def __init__(self, config):
+            super().__init__(config)
+            self.a = torch.nn.Parameter(torch.tensor(config.a).float())
+            self.b = torch.nn.Parameter(torch.tensor(config.b).float())
+
+        def forward(self, input_x, labels=None, **kwargs):
+            y = input_x * self.a + self.b
+            if self.training:
```

Comment:
I don't think this conditional works. I replaced it with `if 1` and the test fails.

It doesn't look this happens?
```
model.train()
```
But I see it `trainer.py` in the right place, alright. Puzzled.

Reply:
wrong diagnosis from my side - the problem was in flakiness - which you have now resolved.